webapp/components/DataQuery/Visualizer/Chart/Chart.js

```diff
@@ -0,0 +1,189 @@
+import React, { useEffect, useState } from 'react'
+import PropTypes from 'prop-types'
+import axios from 'axios'
+import * as A from '@core/arena'
+
+import { useI18n } from '@webapp/store/system'
+import { Query } from '@common/model/query'
+import { useSurveyId, useSurveyCycleKey } from '@webapp/store/survey'
+import PanelRight from '@webapp/components/PanelRight'
+
+export const getUrl = ({ surveyId, query }) => `/api/surveyRdb/${surveyId}/${Query.getEntityDefUuid(query)}/chart`
+
+/*
+ TODO, extract this spec configuration to a pannelRight and manange a Chart class
+
+ three modes, basic [ choose between templates and then allow the user to modify some basic fields ]
+ custom [ customize the charts based on different properties -> the most complex to implement ]
+ advance -> define the spec using a JSON editor
+
+ Maybe the only thing that the user needs to chose on the left pannel is the last entity
+*/
+
+const defaultSpec = {
+  $schema: 'https://vega.github.io/schema/vega-lite/v5.json',
+  /*name: 'tree_health_label',
+  description: 'tree_health_label',
+  title: {
+    text: 'tree_health_label',
+  },*/
+  /*width: 500,
+  height: 500,
+  autosize: {
+    type: 'fit',
+    contatins: 'padding',
+    resize: true,
+  },*/
+  /*data: {
+    name: 'table',
+    values: [],
+  },*/
+
+  layer: [
+    {
+      mark: { type: 'arc', innerRadius: 60, outerRadius: 80 },
+    },
+    /*{
+    mark: {type: "text", radius: 120},
+    encoding: {
+      text: {field: "tree_health_label", type: "nominal"}
+    }
+  }*/
+  ],
+  encoding: {
+    theta: {
+      field: 'tree_health_label',
+      type: 'nominal',
+      aggregate: 'count',
+      impute: {
+        value: 'NULL',
+      },
+      //stack: true
+    },
+    color: {
+      field: 'tree_health_label',
+      type: 'nominal',
+      /*legend: {
+        title: 'Tree health',
+      },*/
+
+      impute: {
+        value: 'NULL',
+      },
+    },
+    /*column: {
+      field: 'tree_origin_label',
+      type: 'nominal',
+      impute: {
+        value: 'NULL',
+      },
+    },
+    row: {
+      field: 'cluster_accessibility_label',
+      type: 'nominal',
+      impute: {
+        value: 'NULL',
+      },
+    },*/
+  },
+}
+
+const updateCategoyOnDonnutChart =
```

Comment:
this function should be move to the chart editor, explained into the panel right comment